core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -318,6 +334,101 @@ private void rename0(Path path, CharSequence tableName, Path otherPath, CharSequ
         }
     }
 
+    public final TelemetryWriterJob startTelemetry() {
+        try {
+            this.telemetryWriterJob = new TelemetryWriterJob(configuration);
+        } catch (SqlException e) {
+            LOG.error().$("starting telemetry failed [error=").$(e.getMessage()).$(']').$();
+            throw CairoException.instance(0).put("Starting telemetry failed");
+        }
+
+        return this.telemetryWriterJob;
+    }
+
+    public final void storeTelemetry(short event) {
+        final MicrosecondClock clock = configuration.getMicrosecondClock();
+        final long cursor = telemetryPubSeq.next();
+        TelemetryRow row = telemetryQueue.get(cursor);
+
+        row.ts = clock.getTicks();
+        row.event = event;
+        telemetryPubSeq.done(cursor);
+    }
+
+    private class TelemetryWriterJob extends SynchronizedJob implements Closeable {
+        private final CharSequence telemetryTableName = "telemetry";
+        private final CharSequence telemetryConfigTableName = "telemetry_config";
+        private final QueueConsumer<TelemetryRow> myConsumer = this::toTelemetryTable;
+        private final StringSink idSink = new StringSink();
+        private final boolean telemetryEnabled = configuration.getTelemetryEnabled();
+        private final TableWriter writer;
+        private final TableWriter writerConfig;
+
+        public TelemetryWriterJob(CairoConfiguration configuration) throws SqlException {
+            sqlExecutionContext.with(AllowAllCairoSecurityContext.INSTANCE, null, null);
+
+            try(Path path = new Path()) {
+                final NanosecondClock nanosecondClock = configuration.getNanosecondClock();
+                final MicrosecondClock microsecondClock = configuration.getMicrosecondClock();
+
+                if (getStatus(AllowAllCairoSecurityContext.INSTANCE, path, telemetryConfigTableName) == TableUtils.TABLE_DOES_NOT_EXIST) {
+                    final Long256Impl id = new Long256Impl();
+                    id.setLong0(nanosecondClock.getTicks());
+                    id.setLong1(microsecondClock.getTicks());
+                    id.toSink(idSink);
+                    compiler.compile("CREATE TABLE " + telemetryConfigTableName + " (id long256, enabled boolean);", sqlExecutionContext);
+                } else {
+                    try (TableReader reader = new TableReader(configuration, telemetryConfigTableName)) {
+                        reader.getCursor().getRecord().getLong256(0, idSink);
+                    }
+                    compiler.compile("TRUNCATE TABLE " + telemetryConfigTableName + ";", sqlExecutionContext);
```

Comment:
Semicolon is optional.

As discussed in-person we probably should keep config table. In fact two:

1. table for guid - created once if it doesn't exist
2. table for ups and downs. I would probably put up/down events to main telemetry table?

Reply:
I don't understand this, this is already the case:

1. We have a table `telemetry_config` that stores `guid` and `enabled`
2. We do exactly that